### DISCUSSION TITLE
DB migrations to a external server

### DISCUSSION THREAD

**Discussion by visuellsoft** [abpframework/abp#8865] [General]
Hello,

I have a question. 
I have no direkt connection to the production server from my developer computer.
I publish the abp-website into a local folder an import the files in the wwwroot on the webserver.
But what is the best way to migrate the database and the migrations in the future? When I develop a (normal) asp.net core app, I get a checkbox to export the migration in a SQL-File and the it is possible to import the migration in the SQL-Management Studio.

Thanks for your support
BR Thomas

**Comment by realLiangshiwei** [comment]
See https://docs.abp.io/en/abp/4.3/Data-Seeding#where-how-to-seed-data

**Reply by visuellsoft** [reply]
Hello,

thank for your answer.
I use DBMigrator always on my developer machine. How can I use DBMigrator on the remote server. My developer machine has not direct connection to the webserver or database. I can connect to the webserver only by Citrix/RDP.

**Comment by 274188A** [comment]
You can use dotnet core cli to generate script

`dotnet ef migrations script`

Also you can put this to file with new power shell out-file command.

`dotnet ef migrations script | out-file ./script.sql`
